scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,396 @@
+
+        sftpPassword = getThirdPartyCredential(sftpUrlKey);
+
+        if (sftpPassword == null) {
+            throw new IllegalArgumentException("Please add your sftp password to 3rd-party credentials under the key :\"" +
+                                               sftpUrlKey + "\"");
+        }
+    }
+
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, JSchException, SftpException {
+
+        List<String> filesRelativePathName = null;
+
+        ChannelSftp channelsftp = null;
+        Session session = null;
+        Channel channel = null;
+
+        try {
+            JSch jsch = new JSch();
+            session = jsch.getSession(sftpUsername, sftpHostname, sftpPort);
+            session.setPassword(sftpPassword);
+            Properties config = new Properties();
+            config.put(STRICT_HOST_KEY_CHEKING, "no");
+            session.setConfig(config);
+            // create sftp Session
+            getOut().println("Connecting to " + sftpUrlKey);
+            session.connect();
+            // open sftp Channel
+            channel = session.openChannel("sftp");
+            channel.connect();
+            channelsftp = (ChannelSftp) channel;
+
+            switch (sftpMode) {
+                //ftp mode is get
+                case GET:
+                    getOut().println("BEGIN DOWNLOAD FROM SFTP SERVER: " + sftpUrlKey);
+                    // recursive folder content download from sftp server
+                    filesRelativePathName = recursiveFolderDownload(channelsftp,
+                                                                    sftpRemoteRelativePath,
+                                                                    sftpLocalRelativePath);
+                    getOut().println("END DOWNLOAD FROM SFTP");
+                    break;
+
+                //ftp mode is put
+                case PUT:
+                    getOut().println("BEGIN UPLOAD TO SFTP SERVER: " + sftpUrlKey);
+                    // Create remote directory(ies) if it(they) does(do) not exist
+                    if (!remoteDirExists(channelsftp, sftpRemoteRelativePath)) {
+                        makeRemoteDirectories(channelsftp, Paths.get(sftpRemoteRelativePath));
+                    }
+                    // recursive folder content upload to sftp server
+                    filesRelativePathName = recursiveFolderUpload(channelsftp,
+                                                                  sftpLocalRelativePath,
+                                                                  sftpRemoteRelativePath);
+                    getOut().println("END UPLOAD TO SFTP");
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("SFTP mode can only be " + PUT + " or " + GET);
+            }
+        } catch (JSchException e) {
+            throw new JSchException("An error occured while trying to connect to SFTP server: ", e);
```

Comment:
Why do you wrap JSchException in JSchException?

Reply:
Just to emit a custom and understandable error message, and forward the stacktrace up...

Reply:
then maybe: `throw new JSchException("An error occured while trying to connect to SFTP server: ", e.getCause());`